dropwizard-auth/src/main/java/io/dropwizard/auth/AuthFilter.java

```diff
@@ -152,7 +152,7 @@ public Principal getUserPrincipal() {
 
                 @Override
                 public boolean isUserInRole(String role) {
-                    return authorizer.authorize(prince, role);
+                    return authorizer.authorize(prince, role, requestContext.getUriInfo());
```

Comment:
Why not pass the complete `ContainerRequestContext` instead of just `UriInfo`?
That way, other use cases, such as using a request header, could also be supported.

Reply:
The main concern I had was that `ContainerRequestContext` allow to change the request in itself, which I'm not sure should be allowed in the `Authorizer`. If that concerns is unfounded, I'm open to make the change.

Reply:
i think passing in the `ContainerRequestContext` is ok, what do you think @joschi ?

Reply:
I made the change in a separate commit.